openapi.yaml

```diff
@@ -132,147 +500,417 @@ securityDefinitions:
 definitions:
-  Meta:
+  PizzaSetResults:
     properties:
-      totalResults:
-        type: integer
-        description: Total number of results
-        example: 10
-      totalPages:
-        type: integer
-        description: Total number of pages
-        example: 10
-      currentPageNumber:
-        type: integer
-        description: Page number of the returned results
-        example: 1
-      currentPageSize:
-        type: integer
-        description: Number of results per page
-        example: 25
-  SelfLink:
+      data:
+        type: array
+        items:
+          $ref: "#/definitions/PizzaResource"
+      links:
+        $ref: "#/definitions/SelfLink"
+  PizzaResult:
     properties:
-      self:
+      data:
+        $ref: "#/definitions/PizzaResource"
+      links:
+        $ref: "#/definitions/SelfLink"
+  PizzaResource:
+    properties:
+      type: &pizzaType
         type: string
-        format: url
-        description: Self-link of current resource
-  PaginationLinks:
+        description: the type of the resource (pizza)
+        enum: [pizza]
+      id: &pizzaId
+        type: string
+        description: a unique ID of a pizza
+        example: abc123
+      links:
+        $ref: "#/definitions/SelfLink"
+      attributes:
+        $ref: "#/definitions/PizzaAttributes"
+      relationships: &pizzaRelationships
+        properties:
+          dough:
+            required: [data]
+            properties:
+              links:
+                properties:
+                  self:
+                    type: string
+                    description: a link to the relationship
+                    example: https://api.oregonstate.edu/v1/pizzas/1/relationships/dough
+                  related:
+                    type: string
+                    description: a link to the dough referenced
+                    example: https://api.oregonstate.edu/v1/pizzas/1/dough
+              data:
+                $ref: "#/definitions/DoughRelationshipResource"
+          ingredients:
+            required: [data]
+            properties:
+              links:
+                properties:
+                  self:
+                    type: string
+                    description: a link to the relationship
+                    example: https://api.oregonstate.edu/v1/pizzas/1/relationships/dough
+                  related:
+                    type: string
+                    description: a link to the ingredients for the dough referenced
+                    example: https://api.oregonstate.edu/v1/pizzas/1/ingredients
+              data:
+                type: array
+                description: a list of relationship resources with the ingredient type
+                items:
+                  $ref: "#/definitions/IngredientRelationshipResource"
+      included: &pizzaIncluded
+        type: array
+        items:
+          type: object
+          description: An ingredient resource or a dough resource
```

Comment:
Can't ingredients be included?

Reply:
Yeah, this list can include both ingredient and dough resources, but there's no way to represent that in the swagger spec at the moment